core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -641,21 +641,25 @@ struct SubviewExtents {
     error(buf + n, buf_len - n, domain_rank + 1, range_rank + 1, dim, args...);
   }
 
+#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
   template <size_t... DimArgs, class... Args>
   KOKKOS_FORCEINLINE_FUNCTION void error(const ViewDimension<DimArgs...>& dim,
                                          Args... args) const {
-#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
```

Comment:
Why did you change this?

Reply:
If `KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST` is not defined, `dim` and `args...` are not used. Since the `(void)` trick doesn't work on parameter packs, this seemed to be the best solution to avoid the warning.